src/lib/pages/Profile.svelte

```diff
@@ -0,0 +1,150 @@
+<script lang="ts">
+  import { Button, ButtonGroup, Input, Spinner } from 'flowbite-svelte';
+  import * as Realm from 'realm-web';
+  import { realmApp } from '../../main';
+  import { Constants, MongoCollections } from '../../Constants';
+  import { onMount } from 'svelte';
+  import type { UserData } from '../models/UserData/UserData';
+  import {
+    Table,
+    TableBody,
+    TableBodyCell,
+    TableBodyRow,
+    TableHead,
+    TableHeadCell,
+  } from 'flowbite-svelte';
+  import { AccountState } from '../enums/AccountState';
+
+  let isLoading = true;
+  const {
+    BSON: { ObjectId },
+  } = Realm;
+
+  onMount(async () => {
+    await getUser();
+  });
+
+  var user: UserData;
+
+  async function getUser() {
+    const mongo = realmApp.currentUser.mongoClient(
+      import.meta.env.VITE_DATA_SOURCE_NAME
+    );
+
+    const data = mongo
+      .db(Constants.DatabaseName)
+      .collection(MongoCollections.Users);
+    const result = (await data.find()) as UserData[];
+    user = result.find((user) => user.personalId === realmApp.currentUser.id);
+    isLoading = false;
+  }
+
+  async function onSaveAll(e: MouseEvent) {
+    const mongo = realmApp.currentUser.mongoClient(
+      import.meta.env.VITE_DATA_SOURCE_NAME
+    );
+
+    const data = await mongo
+      .db(Constants.DatabaseName)
+      .collection(MongoCollections.Users)
+      .updateOne(
+        { personalId: realmApp.currentUser.id },
+        {
+          $set: {
+            name: user.name,
+            surname: user.surname,
+            nationalIdNumber: user.nationalIdNumber,
+            nickname: user.nickname,
+            email: user.email,
+            address: user.address,
+            accountState: 1,
+          },
+        }
+      );
+    getUser();
```

Comment:
Nevrátí se ti to do té data proměnné? Jen aby to nebyl request navíc, jinak taky nemáš await, ten reload dat může proběhnout až za několik vteřin, dobrý by taky bylo setnout loading alespoň